### PR TITLE
Add Seurat functions to R SDK

### PR DESCRIPTION
**Pull request by oligomyeggo**
Add Seurat functions to R SDK, letting users:

- Download their raw Seurat object
- Read their raw Seurat object
- Download their final Seurat object
- Read their final Seurat object (this function will also add a user's custom cluster annotations, and grab a user's custom cluster color palettes to use in external analyses)

### CODE REVIEW

File: R/experiments_data.R [hunk 1]
```diff
@@ -267,3 +267,214 @@ pluto_download_sample_data <- function(experiment_id, dest_filename = NULL){
...
+#' @param silent Boolean, whether to suppress console messages
+#' @returns A list containing `obj`, the requested Seurat object if it was successfully fetched, and `status`, a list containing information from the Pluto API request:\tabular{ll}{
+#'    \code{status_code} \tab HTTP status code (e.g. 200, 400, 401) \cr
+#'    \tab \cr
+#'    \code{code} \tab String, computer-friendly code for response (e.g. `authentication_failed`) \cr
+#'    \tab \cr
+#'    \code{message} \tab Additional details \cr
+#' }
+#' @importFrom utils read.csv
+#' @export
```
Comment (ranikay) [inline R/experiments_data.R:288]: Remove `@importFrom utils read.csv`

File: R/experiments_data.R [hunk 2]
```diff
@@ -267,3 +267,214 @@ pluto_download_sample_data <- function(experiment_id, dest_filename = NULL){
...
+#' @param silent Boolean, whether to suppress console messages
+#' @returns A list containing `obj`, the requested data if it was successfully fetched, and `status`, a list containing information from the Pluto API request:\tabular{ll}{
+#'    \code{status_code} \tab HTTP status code (e.g. 200, 400, 401) \cr
+#'    \tab \cr
+#'    \code{code} \tab String, computer-friendly code for response (e.g. `authentication_failed`) \cr
+#'    \tab \cr
+#'    \code{message} \tab Additional details \cr
+#' }
+#' @importFrom utils read.csv
+#' @export
```
Comment (ranikay) [inline R/experiments_data.R:323]: Remove `@importFrom utils read.csv`

File: R/experiments_data.R [hunk 3]
```diff
@@ -267,3 +267,214 @@ pluto_download_sample_data <- function(experiment_id, dest_filename = NULL){
...
+  seurat_resp <- pluto_GET(url_path_get)
+  if (seurat_type == "raw"){
+    seurat_uuid <- seurat_resp$seurat$items[[1]]$uuid
+  } else if (seurat_type == "final"){
+    seurat_uuid <- seurat_resp$seurat$items[[2]]$uuid
+  } else{
+    stop("Unsupported seurat_type Supported Seurat object types are 'raw', 'final'.")
+  }
+  url_path_post <- paste0("lab/experiments/", experiment_id,
+                          "/files/", seurat_uuid, "/download/")
```
Comment (ranikay) [inline R/experiments_data.R]: Rename to something besides post

**Review by ranikay — CHANGES_REQUESTED** [review]
Tiny fixes, otherwise looks great!

File: R/experiments_data.R (hunk 1 above)
Comment (oligomyeggo) [reply R/experiments_data.R:288]: Done

File: R/experiments_data.R (hunk 2 above)
Comment (oligomyeggo) [reply R/experiments_data.R:323]: Done

File: R/experiments_data.R (hunk 3 above)
Comment (oligomyeggo) [reply R/experiments_data.R]: Done

**Review by ranikay — APPROVED** [review]

**Pull request merged**